Description: Fixes function have_one_kernel in dkms script for non-zero return code
Author: Mario Limonciello <Mario_Limonciello@Dell.com>
Origin: upstream
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1261116

Index: dkms/dkms
===================================================================
--- dkms.orig/dkms	2014-01-10 16:33:48.000168000 +0100
+++ dkms/dkms	2014-01-10 16:44:48.522601051 +0100
@@ -162,11 +162,14 @@
 
 # A little test function for DKMS commands that only work on one kernel.
 have_one_kernel() {
-    (( ${#kernelver[@]} > 1 )) && \
-	die 4 $"The action $1 does not support multiple kernel version" \
-	$"parameters on the command line."
-    [[ $all ]] && die 5 $"The action $1 does not support the --all" \
-	$"parameter."
+    if (( ${#kernelver[@]} != 1 )); then
+        die 4 $"The action $1 does not support multiple kernel version" \
+        $"parameters on the command line."
+    fi
+    if [[ $all ]]; then
+        die 5 $"The action $1 does not support the --all" \
+        $"parameter."
+    fi
 }
 
 # Set up the kernelver and arch arrays.  You must have a 1:1 correspondence --
